src/gearman-client.cpp

```diff
@@ -32,8 +32,26 @@
 
 namespace Driveshaft {
 
-extern "C" {
-static void* worker_callback(gearman_job_st *job, void *context,
+class StringstreamWriter : public Writer {
+public:
+    StringstreamWriter() {
+        // enable exception semantics
+        this->stream.exceptions(std::ios::failbit);
```

Comment:
Note that we weren't setting this before, so the block in curl_write_func() would never have signaled a failed write back to CURL. Hooray for tests!